tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,136 @@
+import copy
+import pytest
+
+from configuration import available_ports, TLS13_CIPHERS, ALL_TEST_CURVES, MINIMAL_TEST_CERTS
+from common import ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version, to_bytes
+
+PADDING_SIZE_SMALL = 250
+PADDING_SIZE_MEDIUM = 1000
+PADDING_SIZE_MAX = 1 << 14
+
+PADDING_SIZES = [
+    PADDING_SIZE_SMALL,
+    PADDING_SIZE_MEDIUM,
+    PADDING_SIZE_MAX
+]
+
+# arbitrarily large payload size
+PAYLOAD_SIZE = 1024
+
+
+def get_record_header(payload_size: int) -> str:
+    # In the TLS record header, the last two bytes are reserved for length
+    hex_string = "{:04x}".format(payload_size)
+    first_byte, second_byte = hex_string[:2], hex_string[2:]
+    return "17 03 03 {} {}".format(first_byte, second_byte)
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+# only tls 1.3 supports record padding
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("padding_size", PADDING_SIZES, ids=get_parameter_name)
+def test_s2n_server_handles_padded_records(managed_process, cipher, provider, curve, protocol, certificate,
+                                           padding_size):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(PAYLOAD_SIZE)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        cipher=cipher,
+        cert=certificate.cert,
+        curve=curve,
+        data_to_send=random_bytes,
+        insecure=True,
+        protocol=protocol,
+        extra_flags=['-record_padding', padding_size]
+    )
+
+    server_options = copy.copy(client_options)
+    server_options.data_to_send = None
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.extra_flags = None
+
+    s2nd = managed_process(S2N, server_options, timeout=5)
+    openssl = managed_process(provider, client_options, timeout=5)
+
+    # expected length is the padding size + 16 bytes of aead tag
+    expected_total_length = padding_size + 16
+    expected_record_header = get_record_header(expected_total_length)
+
+    for client_results in openssl.get_results():
+        client_results.assert_success()
+        # verify that the openssl is sending padded payloads with the expected size
+        assert to_bytes(expected_record_header) in client_results.stdout
```

Comment:
I'm not sure about this as a check. This will pass if the string appears anywhere in the output. That means that it only has to appear in one record, and not even necessarily as the header.

Reply:
Would changing this to `0000 - 17 03 03 {} {}` be a stronger check? This would ensure the 5 bytes we are checking are indeed header bytes

Reply:
That solves the "not even necessarily as the header" problem, but not the "only has to appear in one record" problem. Don't you want to check that all records start with that header / no records start with a different header?

Reply:
Made the check more rigorous. 

FYI, the check isn't as straightforward as "make sure all sent records have size `padding_size + 16`". This is because openssl actually pads a record to the next largest multiple of `padding_size` or `record_size_after_padding = ceiling(payload_size / padding_size) + 16`. For specific records such as the close connection or the session tickets, the payload_size is always less than our tested padding_size but our application data may be larger. 

The change I added does the following:
1. Find all application records sent by openssl
2. For each of those records, ensure that its size is one of the two possible padded sizes and count
3. Verify that we counted the expected number of sent application data + close records or session tickets